docs/DEVELOPMENT.md

```diff
@@ -497,3 +501,56 @@ Summary table of "types" (the first part of the commit message). More details at
 
 Major version bumps are created when `BREAKING CHANGE` is in the commit body or the type is followed by `!` (e.g. `feat!: add a new major feature`). All types can be modified with an additional "scope" as shown in the conventional commit documentation (e.g. `fix(admin): remove broken link in admin panel`).
+
+## Frontend conventions
+
+The following conventions used to build and maintain the frontend UI.
+
+*NOTE: At this time, the current state of the frontend structure and conventions do not follow this documentation. This document is setting standards moving forward.
+
+### Architecture
+
+The frontend UI should be structured to improve consistency and developer experience by establishing and maintaining the following standards. The frontend consists of a single page application architecture using Nunjucks templates to rendered by the server and React to render pages in the browser.
+
+#### Server rendered views (Nunjucks)
+- Templates in the [`./views`](../views) directory.
+
+#### Browser rendered pages (React)
+- Page template components are in the [`./frontend/pages`](../frontend/pages) directory.
+- Only page template components should be used to render the routes in the [`./frontend/routes.js`](../frontend/routes.js) file.
+- Pages are composed of two types of components: Pure components and dynamic components. The majority of components used should be written as pure components.
+- Most HTTP requests should be called from the page component with the response data being passed down to pure components. Dynamic components should be used sparingly and scoped to a specific purpose to encapsulate complexity and state to the specific dynamic component.
+- Most UI state should be defined at the page level using React hooks.
+- Global app state should only be connected to page template components.
+
+#### Components (React)
+
+##### Pure Components
+- Pure components are in the [`./frontend/components`](../frontend/components) directory.
+- These are primarily used to display information or be a general action that can be reused throughout the UI. ie:
+  - A title component that displays the provided string property.
+  - A button component that will call a function on click.
+- All of the component properties are passed directly to the component.
+- Property types should be booleans, numbers, strings, arrays, objects, or functions.
+- No HTTP requests should be called from within a pure component.
+- No state management actions should happen within a pure component unless it is specific to that component with no external side effects. ie:
+  - A menu component with a button that opens an element to display a list of links. The state of the menu being open or closed can be encapsulated within that component and its state does not effect any other components and actions.
+
+##### Dynamic Components
+- Dynamoc components are in the [`./frontend/containers`](../frontend/containers) directory.
```

Comment:
@cloud-gov/pages-ops Let me know if the `./frontend/containers` directory name makes sense for dynamic components. Other frameworks have started using the term `islands`. Let me know if there is another term that is more semantic.